### ISSUE TITLE
SC2216: false positive for piping yes or letter y/n to commands that ask yes/no questions.

### ISSUE THREAD

**Issue by astkaasa** [koalaman/shellcheck#3141]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2216
- My shellcheck version (`shellcheck --version` or "online"):
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

yes | cp -r dist/* .
```

#### Here's what shellcheck currently says:
```sh
$ shellcheck myscript
 
[Line 3:](javascript:setPosition(3, 7))
yes | cp -r dist/* .
      ^-- [SC2216](https://www.shellcheck.net/wiki/SC2216) (warning): Piping to 'cp', a command that doesn't read stdin. Wrong command or missing xargs?

$
```

#### Here's what I wanted or expected to see:
Here `yes` is used to pipe `y` to commands that ask yes/no questions.

Actually `echo` any word begin with `y/n` (case insensitive) should be valid too, but it may be too difficult to be exempted from this rule.

**Comment by BlacksilverConsulting** [comment]
Looking at the docs for cp, ISTM the better way to do this is to use the -f option.

**Comment by koalaman** [comment]
`cp` only asks questions if invoked with `-i`. It's common to have `alias cp='cp -i'` but these do not apply in scripts.

However, shellcheck still warned for `yes | cp -i ...` while handling `cp -i ... < file` correctly. This is now fixed, and `yes | cp -ri dist/* .` should no longer trigger this warning.
